File: wowup-electron/src/app/models/wowup/addon-search-result.ts
import { AddonFundingLink } from "../../entities/addon";
import { AddonSearchResultFile } from "./addon-search-result-file";

export interface AddonSearchResult {
  author: string;
  downloadCount?: number;
  externalId: string;
  externalUrl: string;
  files?: AddonSearchResultFile[];
  name: string;
  providerName: string;
  screenshotUrl?: string;
  screenshotUrls?: string[];
  summary?: string;
  thumbnailUrl: string;
  releasedAt?: Date;
  fundingLinks?: AddonFundingLink[];
  changelog?: string;
}

File: WowUp.Common/Models/Addons/AddonSearchResult.cs
using System.Collections.Generic;

namespace WowUp.Common.Models.Addons
{
    public class AddonSearchResult
    {
        public string Name { get; set; }
        public string Author { get; set; }
        public string ThumbnailUrl { get; set; }
        public string ExternalId { get; set; }
        public string ExternalUrl { get; set; }
        public string ProviderName { get; set; }

        public IEnumerable<AddonSearchResultFile> Files { get; set; }
    }
}
